minimisers/simplex/fMinSearch.m

```diff
@@ -191,7 +199,11 @@
         y(j) = zero_term_delta;
     end
     v(:,j+1) = y;
-    x(:) = y; [f, result] = funfcn(x,varargin{:});
+    if doPlotEvent
+        controls.calcSLD = true;
+    end
+    x(:) = y; [f, result] = funfcn(x, problemStruct, controls, params);
+    controls.calcSLD = false;
```

Comment:
Can the value of `doPlotEvent` change within this loop? If not, it would make sense to move the `controls.calcSLD` lines outside of the loop rather than continually turning on and off.

Reply:
Also, we don't actually do the plot until after the loop, so presumerably we only need to calculate the SLD on the last iteration of the loop. I'm very much not a fan of `if j == n` but . . . It's one to think about, we want to avoid all unnecessary SLD calculations.